### PR TITLE
Fix SecInfo alert filter conditions (master)

### PR DESCRIPTION
**Pull request by timopollmeier**
This fixes various issues in the SecInfo filter conditions and makes them easier to use:
- The condition description now contains the correct count
- The filter placeholders `$f` and `$F` for the email body are now replaced with the condition filter
- The filter condition now only counts new/updated SecInfo depending on the event type
- The filter condition now checks the SecInfo type given in the event data
- New SecInfo is now excluded from the "updated" lists more consistently

**Checklist**:

- Tests N/A
- [x] [CHANGELOG](https://github.com/greenbone/gvmd/blob/master/CHANGELOG.md) Entry

### CODE REVIEW

File: src/manage_sql.c [hunk 1]
```diff
@@ -11824,10 +11825,24 @@ email_secinfo (alert_t alert, task_t task, event_t event,
                                   strlen (list));
     }
 
+  condition_filter = 0;
+  term = NULL;
+  condition_filter_id = alert_data (alert, "condition", "filter_id");
+  if (condition_filter_id)
+    {
+      gchar *quoted_filter_id;
+      quoted_filter_id = sql_quote (condition_filter_id);
```
[the remark this answers is not in the document]
Comment (timopollmeier) [reply src/manage_sql.c]: Done in ee5ab499971c9c5bce7283350a52fe3fa6e1e0e2.

File: src/manage_sql.c [hunk 2]
```diff
@@ -14204,10 +14205,33 @@ alert_secinfo_count (alert_t alert, char *filter_id)
   if (filter_id && strlen (filter_id) && strcmp (filter_id, "0"))
     get.filt_id = filter_id;
 
-  feed_version_epoch = nvts_feed_version_epoch ();
-  db_count = nvt_info_count_after (&get,
-                                   feed_version_epoch,
-                                   get_modified);
+  secinfo_type = alert_data (alert, "event", "secinfo_type");
+  printf ("secinfo_type: %s\n", secinfo_type);
```
[the remark this answers is not in the document]
Comment (timopollmeier) [reply src/manage_sql.c]: Yes, I've removed it in ee5ab499971c9c5bce7283350a52fe3fa6e1e0e2.

File: src/manage_sql.c [hunk 3]
```diff
@@ -14156,6 +14172,76 @@ event_applies (event_t event, const void *event_data,
     }
 }
 
+/**
+ * @brief Return the SecInfo count .
```
[the remark this answers is not in the document]
Comment (timopollmeier) [reply src/manage_sql.c]: Fixed in 041e4bb54bd836761c67a2ad2548955108b5e801.

File: src/manage_sql_secinfo.c [hunk 4]
```diff
@@ -1415,6 +1537,72 @@ DEF_ACCESS (nvt_dfn_cert_adv_iterator_name, 0);
...
+    }
+
+  if (get_modified)
+    extra_where = g_strdup_printf (" AND modification_time > %ld",
+                                   count_time);
+  else
+    extra_where = g_strdup_printf (" AND creation_time > %ld",
+                                   count_time);
+
+  ret = count (type, get, (column_t*)columns, NULL, filter_columns,
```
[the remark this answers is not in the document]
Comment (timopollmeier) [reply src/manage_sql_secinfo.c]: Fixed in 041e4bb54bd836761c67a2ad2548955108b5e801.

File: src/manage_sql.c [hunk 5]
```diff
@@ -14198,28 +14284,10 @@ condition_met (task_t task, report_t report, alert_t alert,
...
-              if (filter_id && strlen (filter_id) && strcmp (filter_id, "0"))
-                get.filt_id = filter_id;
-              db_count = nvt_info_count (&get);
-              if (uuid_was_null)
-                {
-                  free (current_credentials.uuid);
-                  current_credentials.uuid = NULL;
-                }
+              /* SecInfo event. */
+              int db_count;
```
[the remark this answers is not in the document]
Comment (timopollmeier) [reply src/manage_sql.c]: I'm not sure if this exactly, what you meant but I've changed it in 041e4bb54bd836761c67a2ad2548955108b5e801.

[the remark this answers is not in the document]
Comment (mattmundell) [reply src/manage_sql.c]: Yes, that's what I meant, thanks.

**Pull request merged**